### PR TITLE
Add Shotcut

### PR DESCRIPTION
**Pull request by linuxcompitech**
Shotcut is a free, open source, cross-platform video editor. (Qt, GPLv3)

https://www.shotcutapp.com
https://github.com/mltframework/shotcut

### CODE REVIEW

File: srcpkgs/shotcut/template [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Template file for 'shotcut'
+pkgname=shotcut
+version=17.10
+revision=1
+build_style=qmake
+hostmakedepends="qt5-qmake qt5-tools pkg-config"
+makedepends="qt5-declarative-devel qt5-graphicaleffects
+ qt5-multimedia-devel qt5-quickcontrols qt5-webkit-devel
+ qt5-websockets-devel qt5-x11extras-devel
```
Comment (yopito) [inline srcpkgs/shotcut/template]: trailing whitespace (xlint Travis CI job)

File: srcpkgs/shotcut/template [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+pkgname=shotcut
+version=17.10
+revision=1
+build_style=qmake
+hostmakedepends="qt5-qmake qt5-tools pkg-config"
+makedepends="qt5-declarative-devel qt5-graphicaleffects
+ qt5-multimedia-devel qt5-quickcontrols qt5-webkit-devel
+ qt5-websockets-devel qt5-x11extras-devel 
+ mlt-devel mlt-python movit-devel ffmpeg-devel
+ x264-devel libvpx-devel lame-devel
```
Comment (yopito) [inline srcpkgs/shotcut/template]: trailing whitespace (xlint Travis CI job)

File: srcpkgs/shotcut/template [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+ mlt-devel mlt-python movit-devel ffmpeg-devel
+ x264-devel libvpx-devel lame-devel 
+ ladspa-sdk jack-devel libexif-devel
+ gstreamer1-devel SDL_image-devel"
+depends="frei0r-plugins desktop-file-utils"
+short_desc="Free, open source, cross-platform video editor"
+maintainer="Linux CompITech <linuxcompitech@gmail.com>"
+license="GPL-3"
+homepage="https://www.shotcut.org"
+distfiles="https://github.com/mltframework/shotcut/archive/v17.10.tar.gz"
```
Comment (yopito) [inline srcpkgs/shotcut/template]: distfiles="https://github.com/mltframework/shotcut/archive/v${version}.tar.gz"

**Comment by yopito** [comment]
for crossbuild support, try this in your template file (no warranty):
```
checksum=.....

if [ -n "$CROSS_BUILD" ]; then
       hostmakedepends+=" qt5-tools-devel"
fi
```

**Comment by yopito** [comment]
commit message must be "New package: shotcut-17.10" (see  `Manual.md`)

**Comment by newbluemoon** [comment]
At the end of your (cross) build logs is the line:
> Project ERROR: Unknown module(s) in QT: qml quick webkitwidgets multimedia websockets quickwidgets qml-private quick-private x11extras

Which means when cross compiling qt applications you have to find the void packages which contain those modules and add them to `hostmakedepends`. In this case you should add:

    if [ -n "${CROSS_BUILD}" ]; then
    	hostmakedepends+=" qt5-devel qt5-3d-devel qt5-quickcontrols qt5-webkit-devel
    	 qt5-multimedia-devel qt5-websockets-devel qt5-declarative-devel qt5-x11extras-devel"
    fi

**Comment by linuxcompitech** [comment]
OK, I will include end of my template following what you have suggested.

> if [ -n "${CROSS_BUILD}" ]; then
> 	hostmakedepends+=" qt5-devel qt5-3d-devel qt5-quickcontrols qt5-webkit-devel
> 	 qt5-multimedia-devel qt5-websockets-devel qt5-declarative-devel qt5-x11extras-devel"
> fi

**Comment by newbluemoon** [comment]
You need to remove 

    only_for_archs="i686 x86_64"

then, too.

**Comment by newbluemoon** [comment]
When you make changes to your template use `git commit --amend`. That changes your initial commit and doesn’t add a new one eache time you alter your template. There are already a couple of them. Just keep it at the final one. To merge those unecessary commits into one you can use `git rebase -i ...`. [This tutorial](https://help.github.com/articles/using-git-rebase-on-the-command-line/) helped me a lot. :-)

**Comment by cr6git** [comment]
@newbluemoon I'm trying to build the new release `shotcut-17.11` , but cross compilation fails in the same way as [shotcut-17.10](https://travis-ci.org/voidlinux/void-packages/jobs/292589547#L1330). 

````
warning: EVP_CipherFinal is often misused, please use EVP_CipherFinal_ex and EVP_CIPHER_CTX_cleanup
````

Got any ideas that might help?? 
Thank you!

**Comment by newbluemoon** [comment]
@cr6git That’s not the problem. It’s the line below the one you quoted:

```
glwidget.o: In function `Mlt::FrameRenderer::showFrame(Mlt::Frame)':
glwidget.cpp:(.text+0x2a20): undefined reference to `QOpenGLFunctions_1_1::versionProfile()'
```

It means that the function `QOpenGLFunctions_1_1::versionProfile()` is used but its implementation cannot be found in any library. The first thing to do would be finding out where it is hiding.

**Comment by cr6git** [comment]
ok, I see... I guess it's related to [QOpenGLFunctions](http://doc.qt.io/qt-5/qopenglfunctions.html) ?

**Comment by newbluemoon** [comment]
So here’s what I found: `QOpenGLFunctions_1_1::versionProfile()` is part of libQt5OpenGL.so.5.8.0. But seems not to be available on every platform: x86_64 and i686 are ok, it’s missing in the arm* and aarch64 (and mips probably, too) versions. My guess would be that’s because of how libGL is built for these platforms (so that some features are missing when qt is built).

Should the package then use `only_for_archs` in this case?

**Comment by Vaelatern** [comment]
That's scary. Can you confirm it's an arch specific feature, and not a cross build bug?

**Comment by newbluemoon** [comment]
@Vaelatern The only way I can think of right now to do this is to build qt natively on ARM, but I just have a Raspberry Pi 3 and don’t think that’s really feasable.

**Comment by newbluemoon** [comment]
Checking Qt’s latest build logs for [x86_64](https://build.voidlinux.eu/builders/x86_64_builder/builds/6029/steps/shell_3/logs/stdio/text) and [armv7l](https://build.voidlinux.eu/builders/armv7l_builder/builds/5969/steps/shell_3/logs/stdio/text).

x86_64 has
```
g++ -c -pipe -D_FORTIFY_SOURCE=2 -mtune=generic -pipe -DOPENSSL_NO_PSK
-DOPENSSL_NO_NEXTPROTONEG -DOPENSSL_NO_SSL3_METHOD -g -O3 -std=c++1z
-fvisibility=hidden -fvisibility-inlines-hidden -fno-exc    eptions -Wall -W -Wvla -Wdate-time
-Wshift-overflow=2 -Wduplicated-cond -D_REENTRANT -fPIC -DQT_NO_USING_NAMESPACE
-DENABLE_PIXMAN_DRAWHELPERS -DQT_BUILD_GUI_LIB -DQT_BUILDING_QT
-DQT_NO_CAST_TO_A    SCII -DQT_ASCII_CAST_WARNINGS -DQT_MOC_COMPAT
-DQT_USE_QSTRINGBUILDER -DQT_DEPRECATED_WARNINGS
-DQT_DISABLE_DEPRECATED_BEFORE=0x050000 -DQT_NO_EXCEPTIONS -D_LARGEFILE64_SOURCE
-D_LARGEFILE_SOURCE     -DQT_NO_DEBUG -DQT_CORE_LIB -I. -I../../include -I../../include/QtGui
-I../../include/QtGui/5.8.0 -I../../include/QtGui/5.8.0/QtGui -isystem /usr/include/libdrm
-I../../include/QtCore/5.8.0 -I../.    ./include/QtCore/5.8.0/QtCore -I../../include/QtCore -I.moc
-isystem /usr/include/libpng16 -I../../mkspecs/linux-g++ -o .obj/qopenglfunctions_1_1.o
opengl/qopenglfunctions_1_1.cpp
```

But there’s no such line in the armv7l log. Just the header file is installed.

**Comment by cr6git** [comment]
@Vaelatern in this case, would it be an acceptable idea to, temporarily, set `nocross=yes` ?

**Comment by Vaelatern** [comment]
Yes, but `nocross="url://of.failing/build"`

**Comment by cr6git** [comment]
but I have no idea how to do this.

**Comment by cr6git** [comment]
@Vaelatern oh sorry! It's only now I realize that I did misunderstand your last comment.

**Pull request closed without merging**